### DISCUSSION TITLE
Nice write-up of different Nightscout options

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2384] [Show and tell]
@psonnera has done an excellent job of comparing different options on facebook.  This is a copy with his permission:  
  
---  
  
### Migration?
As the deadline is coming closer, one question starts to come up more frequently. What should I do with my Nightscout, should I migrate, where? What is recommended?
There is no universal solution. So, we'll have to read between the lines and understand what's the best strategy for each of us, as we have different needs, skills and budgets.

**HISTORY**
Digging in this group, GitHub and the internet I discovered Jason Calabrese used DigitalOcean, and in the early 2014 the deploy to Azure and Heroku buttons were added to the repo with the original developer's agreement.

**TANSTAAFL**
This is an important concept everybody needs to understand: what we call free means we don't have to pay for it but it also means someone else is paying for it. It doesn't mean it's bad, we all know the price of T1D is already too high for us and our kids but we can't pretend that someone else will pay for it. Whoever offers a free service should have our respect and gratitude, and when they decide the weight of free users is too high and decide to start billing, we should accept it.

**HOSTED SOLUTIONS**
_**You need a reliable Nightscout but don't want to build it and maintain it? T1Pal and NS10BE are your best choices.**_  
Someone told me too few people know about the hosted solutions recommended by our developers. If you will want to pay a vendor for your DIY Nightscout you should seriously consider them. Also, if you need to support multiple T1D people in a family that's surely something to think about. [https://nightscout.github.io/#nightscout-as-a-service](https://nightscout.github.io/?fbclid=IwAR2gDbb_Dh6kkvf97yMGVf_XNRUvknJutSYNU2trVgzzhL9cD8vY0_Z-MnQ#nightscout-as-a-service)

**HEROKU**
_**You liked the Heroku experience and don't want to change? A Hobby account subscription is all it takes to continue like that.**_  
Heroku is a great choice even if 7$ per month is a lot for running Nightscout. Someone reminded me they've been offering a free account to many of us for years so there's no reason not to pay for a Hobby account and let it go as before. Con: it remains DIY, not set it and forget it though. [https://nightscout.github.io/vendors/heroku/hobbyplan/](https://nightscout.github.io/vendors/heroku/hobbyplan/?fbclid=IwAR0oFM9Kl_Avj_gEmshxv5rw_I-U8QXerpFA8hePYBOx9H9UL9q0-kuTxuM)

**MONGODB ATLAS**
_**You need a small free database? An M0 Atlas does that and nothing more.**_  
Since they bought mLab, most of us have a database in MongoDB. This is a weak point of Nightscout DIY and forced our developers to do a lot of work to cope with it. An M0 free database is very limited in size and DIY Loopers/AAPS'ers should be very careful to keep the database below 512MB before their site crashes.

**RAILWAY**
_**You want a simple and free alternative to Heroku? Railway is for you... if you're not worried you might have to migrate again in the future.**_  
We really liked the Railway solution when we found it: it's very simple to migrate and the platform is simple to use. Still, we discovered M5Stack Nightscout devices didn't like the network performance and resets... probably some other followers are suffering related issues. But the main worry with Railway is that they're a startup company and even if they're eager to see more users coming we don't know how they'll react when too many of us use their servers ... for free. TANSTAAFL above. [https://nightscout.github.io/vendors/railway/migration/](https://l.facebook.com/l.php?u=https%3A%2F%2Fnightscout.github.io%2Fvendors%2Frailway%2Fmigration%2F%3Ffbclid%3DIwAR2gDbb_Dh6kkvf97yMGVf_XNRUvknJutSYNU2trVgzzhL9cD8vY0_Z-MnQ&h=AT2GLPnwzKORwSTx02YzJ1JDpsjslBFk30_gcJvjqt0SnAtcA6DmSCiIGU3HrzidtyeI48bTEujjqxin6MOFfN5QPzdj3629Cz1gbgLNWHL_XnMFVbmyknMCtXbn-KsHRPxr&__tn__=-UK-R&c[0]=AT12qnLbaim0MeKQ0DBWKWL1WjMB9dILRwWGh3x44CkgNSBE3e-y-W-D_bEfMwSPEkaWg795AaHwpCyxoLvX9ZZqI6lS4lnIycMg95RIOCrS9eUS9YWtFVxq1Be5KgizqSZdfqGJgQwJ5I5FspNJOVlzVcVTNF2SzWvWmSA)

**NORTHFLANK**
_**Everybody else is going for Railway and you prefer a safer alternative because you're not afraid of an ugly Nightscout site name? Northflank is a good idea.**_  
Not very sponsored, Northflank is also a good option, simple of use and offering a migration utility. The only issue is trying to remember your Nightscout site name (try night--spicy-steam--nightscout--ci95-l9xw.code.run for example). Also, same than above, small company and TANSTAAFL. [https://nightscout.github.io/vendors/northflank/migrate/](https://nightscout.github.io/vendors/northflank/migrate/?fbclid=IwAR0n8Qj9fTsq-qBbtD0ZqVf4FdXGLJz4pIkKh_v_ig_bSlCoVy7zKpjgvvg)

**FLY.IO**
_**You're desperate to migrate now because it's too late for choices and you don't care about the future? Fly.io migration is the easiest immediate choice... but if you don't like typing commands on your computer you'll short be facing a hard stop.**_  
Fly.io was immediately popular due to the ease of migration from Heroku (for those not forgetting to unlink Heroku afterwards). Now, if you want to maintain your site you'll have to use the flyctl command line as the web interface is mainly for monitoring. On the other hand, if you don't mind command line instructions, it is a very good solution. But a computer is mandatory (i.e. issues can't be fixed from your phone) and very few of us go around with a laptop all day long... [https://nightscout.github.io/vendors/fly.io/migrate/](https://nightscout.github.io/vendors/fly.io/migrate/?fbclid=IwAR2eB1mDewLMhjOdpzOc2Df57EiCTf3Q3Yrh9pJVL2qN9nfIu2jQht2BITk)

**AZURE**
_**You need a reliable engine for your Nightscout site, well Azure sure is the way to go. But we've just started evaluating it, so you like taking risks too.**_  
Azure is back hurrah hurrah. Wait, why did we leave it some years ago? Ah, sure, costs. We're working on it, do you want to join the testers? [https://nightscout.github.io/vendors/azure/migrate/](https://l.facebook.com/l.php?u=https%3A%2F%2Fnightscout.github.io%2Fvendors%2Fazure%2Fmigrate%2F%3Ffbclid%3DIwAR0UnFbQPjtioKU6_l6TscjKnO7v2CdNowsZaS2UTayKeVxT7BL8wERp5K0&h=AT3qM9-nnhfoX0mbYlxfQlrgs4ba83cESS93nFl1hsgBfgGLiOx4DI6TXClTVIMtZI6R77PwbC2fSgbHOpOwIVag2APh6qUWirH6pNXr4Wl9RZ7HpQ7IAQJDiOTKKnQ_gk3x&__tn__=-UK-R&c[0]=AT12qnLbaim0MeKQ0DBWKWL1WjMB9dILRwWGh3x44CkgNSBE3e-y-W-D_bEfMwSPEkaWg795AaHwpCyxoLvX9ZZqI6lS4lnIycMg95RIOCrS9eUS9YWtFVxq1Be5KgizqSZdfqGJgQwJ5I5FspNJOVlzVcVTNF2SzWvWmSA)

**VPS: ORACLE, GOOGLE, DIGITALOCEAN, MVPS**
_**You've got your geeky side and shells are behind your keyboard more than on the beach. This is what you need.**_  
Now, let this be false in the near future with the xDrip+ team making it so easy that the solutions above will look difficult, but some more time is needed. Keep an eye on it. [https://navid200.github.io/.../Nightscout/GoogleCloud.html](https://l.facebook.com/l.php?u=https%3A%2F%2Fnavid200.github.io%2FxDrip%2Fdocs%2FNightscout%2FGoogleCloud.html%3Ffbclid%3DIwAR1yHBWbRZq5hTXuQJ-QQG_ZRk72IeZTSVtx9olTk3TSipZEjevjaw_SxyU&h=AT3VnQBJxisAl2K1gQvlaJVDOpPMvl4yP2koFHhn7R65KplSHVMOnWLB0ZD608CmvC10hZoz97OHEO0CEbciXohEmvvhCl_qTwAzfEWuNByME8fHVbpVfAuCSlWk8FUAp8Qf&__tn__=-UK-R&c[0]=AT12qnLbaim0MeKQ0DBWKWL1WjMB9dILRwWGh3x44CkgNSBE3e-y-W-D_bEfMwSPEkaWg795AaHwpCyxoLvX9ZZqI6lS4lnIycMg95RIOCrS9eUS9YWtFVxq1Be5KgizqSZdfqGJgQwJ5I5FspNJOVlzVcVTNF2SzWvWmSA)

**SYNOLOGY NAS**
_**You have one home and know how to hook it safely to the internet, then why not use it and fix that Nightscout problem!**_  
Remember your house network is not as redundant as the cloud... no connection home, no Nightscout... [https://nightscout.github.io/vendors/synology/new_user/](https://nightscout.github.io/vendors/synology/new_user/?fbclid=IwAR3X04EdmvxoTA9_0CHQdmv4f2-UnIGrv9x9VI9gy9wGY6dx9pEwdzsRz_s)  
<br/>  
  
That's about it for now... hopefully we'll have our ideas clearer before the end of next month.  

![311579460_5486429004744257_5450854633888080912_n](https://user-images.githubusercontent.com/51497406/196272909-3896e7f1-85a2-4020-a2bb-dba783e8cb82.jpg)

**Comment by nunosancha** [comment]
![Nightscout Resized](https://github.com/user-attachments/assets/889552f4-75da-479a-ac3a-65852854384c)

Hey there,

Nice article. 

We have a recipe at TRMNL that allows people to display Nightscout data on an e-ink device. And all the options that you mentioned in this article would work with TRMNL.

Once you have a server, you just have to fill in the data:

<img width="843" alt="image" src="https://github.com/user-attachments/assets/19502e8c-3f62-473c-98bf-d0aa04ddc304" />

Just wanted to share this.
